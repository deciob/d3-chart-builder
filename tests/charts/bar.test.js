import {
  select,
} from 'd3-selection';

import jsdom from 'jsdom';

import barChart from '../../src/charts/bar';


test('an horizontal bar chart', () => {
  const { JSDOM } = jsdom;
  const data = [
    { x: 'a', y: -10 },
    { x: 'b', y: 20 },
    { x: 'c', y: 30 },
    { x: 'd', y: 40 },
  ];
  const chart = barChart();
  chart
    .width(800)
    .height(600)
    .transitionDuration(0)
    .transitionStepSeed(0)
    .layout('horizontal')
    .xAccessor(d => d.y)
    .yAccessor(d => d.x)
    .margin({ top: 0, left: 20, right: 10, bottom: 20 });

  const dom = new JSDOM(
    '<div></div>',
  );
  const document = dom.window.document;
  const selection = document.querySelector('div');
  const chartContainer = select(selection);

  const outerHtmlPromise = new Promise((resolve) => {
    chartContainer.datum(data).call(chart);
    setTimeout(() => {
      resolve(dom.window.document.documentElement.outerHTML);
    }, 250);
  });

  return outerHtmlPromise.then(outerHtml => expect(outerHtml).toMatchSnapshot());
});
