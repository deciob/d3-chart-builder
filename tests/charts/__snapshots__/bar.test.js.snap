// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an horizontal bar chart 1`] = `"<html><head></head><body><div><svg width=\\"800\\" height=\\"600\\"><g transform=\\"translate(20, 0)\\"><g class=\\"bars-g\\"><rect class=\\"bar\\" x=\\"0\\" y=\\"15\\" width=\\"154\\" height=\\"127\\"></rect><rect class=\\"bar\\" x=\\"154\\" y=\\"156\\" width=\\"308\\" height=\\"127\\"></rect><rect class=\\"bar\\" x=\\"154\\" y=\\"297\\" width=\\"462\\" height=\\"127\\"></rect><rect class=\\"bar\\" x=\\"154\\" y=\\"438\\" width=\\"616\\" height=\\"127\\"></rect></g><g transform=\\"translate(0, 580)\\" class=\\"x-axis-g\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\"><path class=\\"domain\\" stroke=\\"#000\\" d=\\"M0.5,6V0.5H770.5V6\\"></path><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">-10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(77.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">-5</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(154.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(231.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">5</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(308.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(385.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">15</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(462.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">20</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(539.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">25</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(616.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">30</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(693.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">35</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(770.5,0)\\"><line stroke=\\"#000\\" y2=\\"6\\"></line><text fill=\\"#000\\" y=\\"9\\" dy=\\"0.71em\\">40</text></g></g><g class=\\"y-axis-g\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"end\\"><path class=\\"domain\\" stroke=\\"#000\\" d=\\"M-6,0.5H0.5V580.5H-6\\"></path><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,78.5)\\"><line stroke=\\"#000\\" x2=\\"-6\\"></line><text fill=\\"#000\\" x=\\"-9\\" dy=\\"0.32em\\">a</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,219.5)\\"><line stroke=\\"#000\\" x2=\\"-6\\"></line><text fill=\\"#000\\" x=\\"-9\\" dy=\\"0.32em\\">b</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,360.5)\\"><line stroke=\\"#000\\" x2=\\"-6\\"></line><text fill=\\"#000\\" x=\\"-9\\" dy=\\"0.32em\\">c</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,501.5)\\"><line stroke=\\"#000\\" x2=\\"-6\\"></line><text fill=\\"#000\\" x=\\"-9\\" dy=\\"0.32em\\">d</text></g></g></g></svg></div></body></html>"`;
